### PR TITLE
Refactor clean up tool

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

I used this Skunkwork week to improve our clean-up tool by restructuring the code, making it more reliable and less error-prone. I also changed the flow and the logic behind the cleaning, instead of trying to find and remove the resource in a standalone way, we go through the atlas resource, and any external cloud resource associated gets removed in the flow.

#### Changelist:
* Log all things and never fail when executing
* Flow from Atlas resources to inner cloud provider resources
* Support to clean Atlas Gov
* Add clean-up for Encryption at Rest
* Ass clean-up for teams
* Fancy colored output
* Clean 3h(by default) old resources every 30 minutes from Monday to Friday between 7-22h

There were not many left resources on cloud-qa this week due to skunkworks but there are some test execution on Actions

### CODE REVIEW

**Comment by josvazg** [comment]
Could we summarise in the description the refactor goals?

File: .github/workflows/cleanup-all.yml [hunk 1]
```diff
@@ -1,76 +1,60 @@
...
+        uses: actions/checkout@v4
 
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version-file: "${{ github.workspace }}/go.mod"
+          go-version-file: "${{ github.workspace }}/tools/clean/go.mod"
 
-      - name: Build & Test clean tool
-        run: (cd tools/clean && go build . && go test .)
```
Comment (josvazg) [inline .github/workflows/cleanup-all.yml:44]: Why do we skip self-testing?
This testing will not be run automatically anywhere else AFAIK, which renders the test code pointless.

Comment (josvazg) [reply .github/workflows/cleanup-all.yml:44]: I saw later that tests were removed. Then this makes sense.

On the other hand, this cleanup code is not trivial and lacks any self testing. I guess as it is not production related, we can relax and react if that becomes a problem at some point.

**Review by josvazg — COMMENTED** [review]
LGTM, but I would like to see links to tests of these cleanup refactor working before approval.

**Review by josvazg — APPROVED** [review]
LGTM!

Thanks for the tests and summary:
✅ https://github.com/mongodb/mongodb-atlas-kubernetes/actions/runs/6657764330

**Review by s-urbaniak — APPROVED** [review]

**Review by igor-karpukhin — APPROVED** [review]
Great job! 👍

**Pull request merged**